### PR TITLE
Update gmm_ops.py

### PR DESCRIPTION
**Pull request by gautam1858**
It should be int64 instead of int32

### CODE REVIEW

File: tensorflow/contrib/factorization/python/ops/gmm_ops.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def _init_clusters_random(data, num_clusters, random_seed):
         maxval=math_ops.cast(num_data, dtypes.int64),
         seed=random_seed,
         dtype=dtypes.int64)
-  indices = math_ops.cast(indices, dtypes.int32) % num_data
+  indices = math_ops.cast(indices, dtypes.int64) % num_data
```
Comment (agarwal-ashish) [inline tensorflow/contrib/factorization/python/ops/gmm_ops.py]: indices is already int64. You can remove this cast.

**Review by martinwicke — APPROVED** [review]
I addressed Ashish's comments.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
The relevant tests fail with the cast removed. @agarwal-ashish is this expected?

**Comment by drpngx** [comment]
Jenkisn, test this please.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
@agarwal-ashish could you take a look at the failing tests?

**Comment by agarwal-ashish** [comment]
Please refer to my earlier comment. You need to cast num_data to int64 in
gmm_ops.py.

**Comment by drpngx** [comment]
Sorry @agarwal-ashish , I meant @gautam1858 . Thanks for the hint!

**Comment by gautam1858** [comment]
what should I do ?

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
```
CMakeFiles/iron_validatable_behavior.dir/build.make:89: recipe for target 'iron_validatable_behavior/src/iron_validatable_behavior-stamp/iron_validatable_behavior-download' failed
make[2]: *** [iron_validatable_behavior/src/iron_validatable_behavior-stamp/iron_validatable_behavior-download] Error 1
CMakeFiles/Makefile2:7918: recipe for target 'CMakeFiles/iron_validatable_behavior.dir/all' failed
make[1]: *** [CMakeFiles/iron_validatable_behavior.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Seems like a transient download error.

Jenkins, test this please.

**Pull request merged**